src/persist-client/src/internal/trace.rs

```diff
@@ -984,17 +984,18 @@ struct Spine<T> {
 
 impl<T> Spine<T> {
     pub fn map_batches<'a, F: FnMut(&'a SpineBatch<T>)>(&'a self, mut f: F) {
-        for batch in self.merging.iter().rev() {
-            match batch {
-                MergeState::Double(batch1, batch2, _) => {
-                    f(batch1);
-                    f(batch2);
-                }
-                MergeState::Single(batch) => f(batch),
-                _ => {}
-            }
+        for batch in self.spine_batches() {
+            f(batch);
         }
     }
+
+    pub fn spine_batches(&self) -> impl Iterator<Item = &SpineBatch<T>> {
```

Comment:
Clever!

Reply:
I stole this trick from `MfpPlan::evaluate`, so credit goes to Frank I think!